uint8 x
uint8 y
uint8 n
uint8 index
IntersectionLine[] int_lines
